internal/config/config.go

```diff
@@ -0,0 +1,101 @@
+// Package config handles environment-based configuration loading with sensible defaults
+package config
+
+import (
+	"os"
+	"strconv"
+	"time"
+)
+
+// Singleton instance of configuration
+var appConfig *Config = nil
```

Comment:
```suggestion
var appConfig *Config
```

Is there any difference?

Reply:
Not really, pointer will anyhow default to nil upon declaration... why it got there? I'm guessing muscle memory from Python where one has to actually assign value to a variable in order to declare it and access it for the first time before lazy-initialize it.

Reply:
Correct — but it’s due to one of Go’s key features: variables in Go can’t be uninitialized thanks to its _zero-value mechanism_.